third_party/sbedecoder/message.py

```diff
@@ -126,6 +142,40 @@ def __str__(self, raw=False):
             return f'{self.name}: {str(self.value)} ({str(self.raw_value)}'
         return f'{self.name}: {str(self.value)}'
 
+
+    @staticmethod
+    def get_json_field_type(part: DatacastField = None):
+        field = part
+        json_type = {}
+        json_type[field.name] = {}
+
+        if isinstance(field, TypeMessageField):
+            if field.is_bool_type is True:
+                json_type[field.name]['type'] = 'boolean'
+            else:
+                mapped_type = json_type_map[field.primitive_type]
+                json_type[field.name]['type'] = mapped_type
+        elif isinstance(field, EnumMessageField):
+            if field.is_bool_type is True:
+                json_type[field.name]['type'] = 'boolean'
+            else:
+                json_type[field.name]['type']  = 'string'
+        elif isinstance(field, SetMessageField):
+            json_type[field.name]['type'] = 'string'
+        else:
+            logging.warning('Unknown type for field: %s', field.name)
+        return json_type
```

Comment:
Is there any harm in not setting a type? Should it be defaulted to string with the warning?

Reply:
defensively returning a string in the latest commit, we might have to reconsider the approach based on broader testing